Content.Client/Construction/UI/ConstructionMenuPresenter.cs

```diff
@@ -185,40 +205,50 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
             // There is apparently no way to set which
         }
 
-        private void PopulateCategories()
+        private void PopulateCategories(string? selectCategory = null)
         {
             var uniqueCategories = new HashSet<string>();
 
-            // hard-coded to show all recipes
-            uniqueCategories.Add("construction-category-all");
-
             foreach (var prototype in _prototypeManager.EnumeratePrototypes<ConstructionPrototype>())
             {
                 var category = prototype.Category;
 
                 if (!string.IsNullOrEmpty(category))
                     uniqueCategories.Add(category);
             }
+            var sortedCategories = uniqueCategories.OrderBy(x => Loc.GetString(x)).ToList();
+            // hard-coded to show all recipes
+            sortedCategories.Insert(0, _forAllCategoryName);
 
-            _constructionView.Category.Clear();
+            if (_favoritedRecipes.Count > 0)
+            {
+                // hard-coded to show favorites if it need
+                sortedCategories.Insert(1, _favoriteCatName);
+            }
 
-            var array = uniqueCategories.OrderBy(Loc.GetString).ToArray();
-            Array.Sort(array);
+            var categoriesArray = sortedCategories.ToArray();
+            _constructionView.Category.Clear();
```

Comment:
This seems entirely unnecessary when you can just make sortedCategories an array?